src/util/astWalker.js

```diff
@@ -0,0 +1,38 @@
+'use strict'
+/*
+  Crawl the given AST through the function walk(ast, callback)
+*/
+function AstWalker () {
+}
+/*
+    * visit all the AST nodes
+    *
+    * @param {Object} ast  - AST node
+    * @param {Object or Function} callback  - if (Function) the function will be called for every node.
+    *                                       - if (Object) callback[<Node Type>] will be called for
+    *                                         every node of type <Node Type>. callback["*"] will be called fo all other nodes.
+    *                                         in each case, if the callback returns false it does not descend into children.
+    *                                         If no callback for the current type, children are visited.
+*/
+AstWalker.prototype.walk = function (ast, callback) {
+  if (manageCallBack(ast, callback) && ast.children && ast.children.length > 0) {
+    for (var k in ast.children) {
+      var child = ast.children[k]
+      this.walk(child, callback)
+    }
+  }
+}
+
+function manageCallBack (node, callback) {
+  if (callback instanceof Function) {
```

Comment:
We could do the following inside `walk`:
```
if (callback instanceof Function) {
  callback = {'*': callback}
}
```
and eliminate the need for performing the check for every node here.

Perhaps also:
```
if (!('*' in callback)) {
  callback['*'] = function() { return true }
}
```